src/components/molecules/LoginForm/LoginForm.tsx

```diff
@@ -0,0 +1,139 @@
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { useState, FC, FormEvent, ChangeEvent } from 'react';
+
+import Button, { EButtonTheme } from '@components/atoms/Button';
+import Input from '@components/atoms/Input';
+import { ETypographyVariant } from '@enums/typography';
+import { useAuth } from '@src/contexts/AuthContext';
+import { Auth } from '@types';
+
+import * as S from './LoginForm.styled';
+
+interface LoginProps {
+  type: string;
+}
+
+const LoginForm: FC<LoginProps> = ({ type }) => {
+  const { login, signup } = useAuth();
+  const router = useRouter();
+  const [loginData, setLoginData] = useState<Auth.ILoginData>({
+    email: '',
+    password: '',
+  });
+  const [signUpData, setSignUpData] = useState<Auth.ISignUpData>({
+    email: '',
+    password: '',
+    repeatPassword: '',
+  });
+
+  const handleLogin = async (e: FormEvent) => {
+    e.preventDefault();
+    try {
+      await login(loginData.email, loginData.password);
+      router.push('/');
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  const handleSignup = async (e: FormEvent) => {
+    e.preventDefault();
+    try {
+      await signup(signUpData.email, signUpData.password);
+      router.push('/');
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  return (
+    <S.FormWrapper>
+      <S.Subheading as="h1" displayAs={ETypographyVariant.H1}>
+        {type === 'login' ? 'login' : 'sign up'}
```

Comment:
If you use the enum you can then replace some of the hard coded values
```suggestion
        {type === EAuthType.Login ? 'login' : 'sign up'}
```